### ISSUE TITLE
Use MongoDB generated IDs instead of deployd generated ID

### ISSUE THREAD

**Issue by NicolasRitouet** [deployd/deployd#766]
To improve migration, export and work with other tools, we should use mongodb ids instead of generating our own within deployd.

This will be a breaking change with all existing deployd instance.

**Comment by hjanuschka** [comment]
so this would mean that `this.id` is not a string anymore insteat  something like `ObjectId("507f191e810c19729de860ea")` ? 

can we generate the id in advance? so e know it before the actual insert?
could the event get a default `self.id=ObjectId("507f191e810c19729de860ea").str` so its backward compatible?

**Comment by NicolasRitouet** [comment]
@hjanuschka we don't even know if we're gonna do it yet. We can't tell how it's gonna look like yet.
One thing for sure is that we'll make all we can to make the transition as smooth as possible.

**Comment by hjanuschka** [comment]
@NicolasRitouet many thx for your hard work and 👍  for trying to make a smooth transition.
what is the exact advantage? - if i export DPD data and import it to any other mongo db - i just prefix my _id with "DPD" - to stay uniqe ;)

i don't need to have multi datasource functionallity - because i think this would break alot of our `beforerequest` mongo-query tuning.

let me know if i can help in anyway, or provide some more details about our usage, i just keep an eye on it as we launched a few months ago, a real heavy desktop, ios,android app - wich is all dpd backed :)

**Comment by andreialecu** [comment]
Even if this gets done it shouldn't necessarily be a breaking change because it would only apply to new rows being inserted. Old rows can keep the old style id and still work.

A true Mongo objectid can be generated ahead of time it seems. So it shouldn't make any difference for anything else either.

**Comment by houmark** [comment]
I'd support this if it's completely non-breaking in a way the ID is being delivered as text no matter how it's stored in MongoDB when doing `this.id` and it's only affecting new inserts.

For some people it may still be breaking because they rely on the length of the ID in some situations, but that should be noted in the release.

I think the thought of being able to run another datasource will require more or less a full rewrite of deployd, so with this change, deployd is MongoDB and MongoDB only.

**Comment by andreialecu** [comment]
The change would be in the `database` layer which is written for mongodb now, and it would be alongside other mongodb specific code. It won't be deeply rooted into deployd anywhere else, and yes, the ids are  still strings basically, just a little longer.

**Comment by houmark** [comment]
Right, another nice thing about true MongoDB ids is that it has natural sorting build in so by sorting on the ID field you automatically get them in order because mongoDB keeps an unix time as a part of the object which is then used for sorting.

**Comment by amitsang2000** [comment]
But the issue is after the importer imports the MongoDB object_id as the id of the rows, the collection doesn't gets updated when we try to edit the collection through deployd dashboard. It shows the error "no object exists that matches the id / query"
